frameworks/elastic/tests/test_backup_restore.py

```diff
@@ -0,0 +1,109 @@
+    key_id = os.getenv("AWS_ACCESS_KEY_ID")
+    if not key_id:
+        assert (
+            False
+        ), 'AWS credentials are required for this test. Disable test with e.g. TEST_TYPES="sanity and not aws"'
+    master_0_node_id = sdk_tasks.get_task_ids(config.SERVICE_NAME, "master-0-node")
+    task_list = sdk_tasks.get_task_ids(config.SERVICE_NAME)
+    for id in task_list:
+        sdk_cmd.run_cli(
+            "task exec {} bash -c \"export JAVA_HOME=\$(ls -d \$MESOS_SANDBOX/jdk*/);  echo '{}' | ./elasticsearch-*/bin/elasticsearch-keystore add --stdin s3.client.default.access_key\"".format(
+                id, os.getenv("AWS_ACCESS_KEY_ID")
+            )
+        )
+        sdk_cmd.run_cli(
+            "task exec {} bash -c \"export JAVA_HOME=\$(ls -d \$MESOS_SANDBOX/jdk*/);  echo '{}' | ./elasticsearch-*/bin/elasticsearch-keystore add --stdin s3.client.default.secret_key\"".format(
+                id, os.getenv("AWS_SECRET_ACCESS_KEY")
+            )
+        )
+        sdk_cmd.run_cli(
+            "task exec {} bash -c \"export JAVA_HOME=\$(ls -d \$MESOS_SANDBOX/jdk*/);  echo '{}' | ./elasticsearch-*/bin/elasticsearch-keystore add --stdin s3.client.default.session_token\"".format(
+                id, os.getenv("AWS_SESSION_TOKEN")
+            )
+        )
+        sdk_cmd.run_cli(
+            'task exec {} bash -c "export JAVA_HOME=\$(ls -d \$MESOS_SANDBOX/jdk*/); ./elasticsearch-*/bin/elasticsearch-keystore list"'.format(
+                id
+            )
+        )
+
+    sdk_cmd.run_cli(
+        "task exec {} /opt/mesosphere/bin/curl -i -XPOST -H 'Content-type: application/json' \"http://coordinator.elastic.l4lb.thisdcos.directory:9200/_nodes/reload_secure_settings\"".format(
+            master_0_node_id[0]
+        )
+    )
+    sdk_cmd.run_cli(
+        "task exec "
+        + master_0_node_id[0]
+        + '  /opt/mesosphere/bin/curl -i -XPOST -H \'Content-type: application/json\' -d \'{"name": "Niharika"}\' "http://coordinator.elastic.l4lb.thisdcos.directory:9200/customer/entry/99?pretty"'
+    )
+    sdk_cmd.run_cli(
+        "task exec "
+        + master_0_node_id[0]
+        + ' /opt/mesosphere/bin/curl -i -XPUT -H \'Content-type: application/json\' -d \'{"type": "s3", "settings": {"bucket": "elastic-bkp-bucket", "region": "us-east-1"} }\' "http://coordinator.elastic.l4lb.thisdcos.directory:9200/_snapshot/s3_repo?verify=false&pretty"'
+    )
+
+    # take backup
+    sdk_cmd.run_cli(
+        "task exec {} /opt/mesosphere/bin/curl -i -XPUT -H 'Content-type: application/json' \"http://coordinator.elastic.l4lb.thisdcos.directory:9200/_snapshot/s3_repo/snap1?\"".format(
+            master_0_node_id[0]
+        )
+    )
+
+    # Delete data before executing restore
+    sdk_cmd.run_cli(
+        "task exec {} /opt/mesosphere/bin/curl -i -XDELETE -H 'Content-type: application/json' \"http://coordinator.elastic.l4lb.thisdcos.directory:9200/*\"".format(
+            master_0_node_id[0]
+        )
+    )
```

Comment:
We should verify that the data really isn't there after this step.

Reply:
Please find this in new PR #68.